### PR TITLE
Disallow mismatched ground truth / prediction label keys for classification tasks

### PR DESCRIPTION
**Pull request by ntlind**
## Improvements
- For classification tasks, throw an error if our prediction label keys don't match our ground truth label keys for all datums
- Fix existing test cases which didn't meet the criteria above

### CODE REVIEW

File: api/tests/functional-tests/crud/test_create_delete.py [hunk 1]
```diff
@@ -1316,13 +1316,8 @@ def test_create_clf_metrics(
     )
     assert len(resp) == 1
     resp = resp[0]
-    missing_pred_keys = resp.missing_pred_keys
```
Comment (ntlind) [inline api/tests/functional-tests/crud/test_create_delete.py:1319]: these values are always empty lists for classification tasks now, so these tests aren't necessary

File: integration_tests/conftest.py [hunk 2]
```diff
@@ -776,17 +776,19 @@ def pred_semantic_segs(
...
+    model_name: str, img5: Datum, img6: Datum, img8: Datum
+) -> list[Prediction]:
     return [
         Prediction(
             datum=img5,
             annotations=[
                 Annotation(
                     task_type=TaskType.CLASSIFICATION,
                     labels=[
-                        Label(key="k12", value="v12", score=0.47),
```
Comment (ntlind) [inline integration_tests/conftest.py:787]: the existing tests violated our new validation check, so I had to rewrite these test values and downstream assertions

File: api/valor_api/backend/core/evaluation.py [hunk 3]
```diff
@@ -320,15 +320,10 @@ def _create_responses(
...
-                    datum_filter,
-                    model_filter,
-                    label_map=parameters.label_map,
-                )
+                # classification tasks have to have matching sets of prediction label keys and ground truth label keys
                 kwargs = {
-                    "missing_pred_keys": missing_pred_keys,
-                    "ignored_pred_keys": ignored_pred_keys,
+                    "missing_pred_keys": [],
+                    "ignored_pred_keys": [],
```
Comment (czaloom) [inline api/valor_api/backend/core/evaluation.py:326]: @ekorman are you happy with this change?

**Review by czaloom — CHANGES_REQUESTED** [review]

**Comment by czaloom** [comment]
@ntlind could you make sure that the tabular classification notebook still runs?

File: api/tests/functional-tests/crud/test_create_delete.py [hunk 4]
```diff
@@ -342,9 +342,9 @@ def pred_clfs_create(
                 schemas.Annotation(
                     task_type=enums.TaskType.CLASSIFICATION,
                     labels=[
-                        schemas.Label(key="k2", value="v2", score=1.0),
-                        schemas.Label(key="k3", value="v3", score=0.87),
-                        schemas.Label(key="k3", value="v0", score=0.13),
+                        schemas.Label(key="k2", value="v2", score=0.8),
```
[the remark this answers is not in the document]
Comment (ntlind) [reply api/tests/functional-tests/crud/test_create_delete.py:345]: yes. k4 and k3 don't exist on these ground truths, so they can't exist on the predictions without throwing the ValueError

File: api/valor_api/backend/core/evaluation.py (hunk 3 above)
Comment (ntlind) [reply api/valor_api/backend/core/evaluation.py:326]: these lists will necessarily be empty now that we're guaranteeing that the prediction label key set matches the ground truth label key set for all datums

Comment (ekorman) [reply api/valor_api/backend/core/evaluation.py:326]: so should we not return these anymore?

**Review by ekorman — APPROVED** [review]

**Review by czaloom — APPROVED** [review]

**Pull request merged**